waspc/data/Cli/templates/basic/.gitignore

```diff
@@ -1,3 +1,11 @@
 /.wasp/
-/.env.server
-/.env.client
+
+# We ignore env files recognized and used by Wasp.
+.env.server
+.env.client
+
+# To be extra safe, we by default ignore any files with `.env` extension in them.
```

Comment:
I would avoid doing this and only take care of the Wasp specific `.env` files.

Next.js's gitignore: https://github.com/vercel/next.js/blob/canary/packages/create-next-app/templates/default/ts/gitignore
Vite React's gitignore: https://github.com/vitejs/vite/blob/main/packages/create-vite/template-react-ts/_gitignore
Remix: https://github.com/remix-run/remix/blob/main/templates/remix/.gitignore

None of the above try to "protect" users that much. I would personally avoid doing this.

Reply:
What about giving users the option for easy switch in .gitignore? That way you aren't too protective of users, but you offer them a way to include it themselves without too much hassle.

For example:

#For ignoring .env files, uncomment the lines below. 
#.env.server
#.env.client

Reply:
Interesting that they don't, but what are the arguments for not doing so?

I quite like the solution in this PR, as it ensures that credentials are not submitted by accident. You can easily name your file `.env.sevrer` instead of `.env.server` and then have quite some problems. Or maybe you name it `.server.env` by accident.
@matijaSos had unpleasant experience where he put `.env.server` in the wrong location and committed it by accident -> if Sengrid's bot hasn't detected it, it would stay up there for who knows how long. And he has to doctor the branch then, change API keys, ... .

Wasp is about best practices, and the best default is not to commit any .env files -> that is why we would have this .gitignore .
The only "bad" side of it is if you really want to commit something that has `.env` in it. That won't happen often though, maybe once or twice for the lifetime of the project, and it will likely be something like `example.env.server`. So the "bad" thing then is that it will be ignored by default, and user will have to realize it is due to .gitignore file and edit it, ideally by adding `!example.env.server` to it. I see that as a minor inconvenience, and a good trade off, in order to have a piece of mind to not by accident commit a dot env file.

Reply:
I understand your position and agree that Wasp is about best practices. I've tried to lay out some of the industry leading frameworks' preferences when it comes to the `.gitignore` file. 

I've stated my personal preference not to do it, but I won't push the issue too hard. This is also okay 👍

Reply:
> 

That is interesting idea but I would rather go the opposite -> give them an easy way to opt out of protection.

Btw for .env.client and .env.server we certainly want to do the ignoring.

The question is though do we want to do it for everything that has .env in it.

Reply:
yep, I agree, we saw in practice in happened multiple times already. With wasp it's a bit tricky because its not obvious where should the env file be. If we had only one place where it makes sense it is then I'd say it's ok to go with relaxed version, but in practice this has shown to be a problem.

So I'd rather be more restrictive for now, and it's very unlikely IMO that somebody will want to upload their env files.

Reply:
Ok, good feedback everybody, we decided to give this more defensive approach a try, and let's see what people will say!